File: pokemon-ng/src/app/classes/trainer/trainer.ts
import {Byte} from "@angular/compiler/src/util";
import {PokemonDTO} from "../pokemon-dto/pokemon-dto";

export class Trainer {
  constructor(
    private _name: string,
    private _birthDate: Date,
    private _hobby: string,
    private _pictureUrl: string,
    private _pokemonList: [],
    private _id?: number,
    private _pokemons: PokemonDTO[] = []
  ) {}


  get pokemons(): PokemonDTO[] {
    return this._pokemons;
  }

  set pokemons(value: PokemonDTO[]) {
    this._pokemons = value;
  }

  public get pokemonList(): [] {
    return this._pokemonList;
  }
  public set pokemonList(value: []) {
    this._pokemonList = value;
  }

  public get pictureUrl(): string {
    return this._pictureUrl;
  }

  public set pictureUrl(value: string) {
    this._pictureUrl = value;
  }

  get id(): number {
    return !this._id ? -1 : this._id;
  }

  get name(): string {
    return this._name;
  }

  get birthDate(): Date {
    return this._birthDate;
  }

  get hobby(): string {
    return this._hobby;
  }

  set id(id: number) {
    this._id = id;
  }

  set name(name: string) {
    this._name = name;
  }

  set birthDate(birthDate: Date) {
    this._birthDate = birthDate;
  }

  set hobby(hobby: string) {
    this._hobby = hobby;
  }

  addPokemon(pokemon : PokemonDTO) {
    this._pokemons.push(pokemon);
  }

}

File: pokemon-java/src/main/java/com/ironhack/pokemon/model/Trainer.java
package com.ironhack.pokemon.model;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
public class Trainer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private LocalDate birthDate;
    @Column(columnDefinition="TEXT")
    private String pictureUrl;
    private String hobby;

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinTable(
            name = "trainer_pokemon",
            joinColumns = {@JoinColumn(name = "trainer_id")},
            inverseJoinColumns = {@JoinColumn(name = "pokemon_id")}
    )
    private List<Pokemon> pokemonList;



    public Trainer() {
    }

    public Trainer(String name, LocalDate birthDate, String pictureUrl, String hobby) {
        this.name = name;
        this.birthDate = birthDate;
        this.pictureUrl = pictureUrl;
        this.hobby = hobby;
    }

    public Trainer(String name, LocalDate birthDate, String pictureUrl, String hobby, List<Pokemon> pokemonList) {
        this.name = name;
        this.birthDate = birthDate;
        this.pictureUrl = pictureUrl;
        this.hobby = hobby;
        this.pokemonList = pokemonList;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LocalDate getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate = birthDate;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public void setPictureUrl(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }

    public String getHobby() {
        return hobby;
    }

    public void setHobby(String hobby) {
        this.hobby = hobby;
    }

    public List<Pokemon> getPokemonList() {
        return pokemonList;
    }

    public void setPokemonList(List<Pokemon> pokemonList) {
        this.pokemonList = pokemonList;
    }
}
